up-l1/README.adoc

```diff
@@ -230,14 +247,41 @@ registerListener(sourceFilter: UUri, sinkFilter: UUri [0..1], listener: UListene
 This API is used to implement the _push_ <<delivery-method, delivery method>> on top of the underlying communication protocol.
 After this method has completed successfully, the given listener will be invoked for each message that matches the given source and sink filter patterns according to the rules defined by the link:../basics/uri.adoc[UUri specification].
 
-UTransport implementations
-
+[.specitem,oft-sid="req~utransport-registerlistener-error-unimplemented~1",oft-needs="impl,utest"]
+--
 * *MUST* fail with a `UCode.UNIMPLEMENTED` if the transport does not support the _push_ <<delivery-method, delivery method>>. In that case, the <<unregister-listener, unregisterListener>> method *MUST* also fail accordingly.
+--
+
+[.specitem,oft-sid="req~utransport-registerlistener-error-resource-exhausted~1",oft-needs="impl,utest"]
+--
 * *MUST* fail with a `UCode.RESOURCE_EXHAUSTED`, if the maximum number of listeners is reached
+--
+
+[.specitem,oft-sid="req~utransport-registerlistener-error-invalid-parameter~1",oft-needs="impl,utest"]
+--
+* *MUST* fail with a `UCode.INVALID_ARGUMENT` if the passed UUri failed validation
+--
```

Comment:
FMPOV this is already covered by validation of the message

Reply:
registerListener() API does not get passed a UMessage so we still need to validate the passed URIs used in the filters.

Reply:
yes, my bad, I thought it was in the _send_ section ...